### ISSUE TITLE
Where to get facenet.pb, mobilenet.pb files ???

### ISSUE THREAD

**Issue by RonakDedhiya** [oylz/DS#9]
Hello,  I was trying to use deep features and want to know which facenet and mobilenet model did u used. Which would be compatible with tensorflow version. If you have it, can you pass out the link?

**Comment by RonakDedhiya** [comment]
when i tried using facenet model. Model is getting loaded but session get into some error:
What eigen library version would be suitable to use?
Sources suggest that some initialization is not done or size is exceeding limit which causing this error? Any Suggestions ??

node name:InceptionResnetV1/Bottleneck/BatchNorm/Reshape_1
node name:embeddings/Square
node name:embeddings/Sum/reduction_indices
node name:embeddings/Sum
node name:embeddings/Maximum/y
node name:embeddings/Maximum
node name:embeddings/Rsqrt
node name:embeddings
session.run----rcs.size():1, ftm2-ftm1:1276216
DS: /home/aitoe/.cache/bazel/_bazel_aitoe/84ac4d8c70e16781a4920ba7d9cd7a15/external/eigen_archive/Eigen/src/Core/DenseCoeffsBase.h:425: Eigen::DenseCoeffsBase<Derived, 1>::Scalar& Eigen::DenseCoeffsBase<Derived, 1>::operator()(Eigen::Index) [with Derived = Eigen::Matrix<float, 1, 128, 1>; Eigen::DenseCoeffsBase<Derived, 1>::Scalar = float; Eigen::Index = long int]: Assertion `index >= 0 && index < size()' failed.
Aborted (core dumped)

**Comment by oylz** [comment]
Sorry, It seems like the returned vector(fts) size is not 128. may be the model of FaceNet has been changed at now.
As you can see, FaceNet  module updatime is:20180402, later than I used it.

**Comment by oylz** [comment]
It seems like the returned vector size is not 128. may be the model of FaceNet has been changed at now.
As you can think, FaceNet  module updatime is:20180402, later than I used it.

you can printf the [len] variable whitch in:https://github.com/oylz/DS/blob/master/deepsort/FeatureGetter/FeatureGetter.cpp#L148
if it's not 128, Can be sure that the model has changed.

**Comment by oylz** [comment]
@RonakDedhiya 
It seems like the returned vector size is not 128. may be the model of FaceNet has been changed at now.
As you can think, FaceNet  module updatime is:20180402, later than I used it.

you can printf the [len] variable whitch in:https://github.com/oylz/DS/blob/master/deepsort/FeatureGetter/FeatureGetter.cpp#L148
if it's not 128, Can be sure that the model has changed.

**Comment by oylz** [comment]
@RonakDedhiya 
It seems like the returned vector size is not 128. may be the model of FaceNet has been changed at now.
As you can think, FaceNet  module updatime is:20180402, later than I used it.

you can printf the [len] variable whitch in:https://github.com/oylz/DS/blob/master/deepsort/FeatureGetter/FeatureGetter.cpp#L148
if it's not 128, Can be sure that the model has changed.


I should tell you, the FaceNet model is not good for tracking, as mine testing.

**Comment by oylz** [comment]
It seems like the returned vector size is not 128. may be the model of FaceNet has been changed at now.
As you can think, FaceNet module updatime is:20180402, later than I used it.

you can printf the [len] variable whitch in:https://github.com/oylz/DS/blob/master/deepsort/FeatureGetter/FeatureGetter.cpp#L148
if it's not 128, Can be sure that the model has changed.

I should tell you, the FaceNet model is not good for tracking, as mine testing.

**Comment by RonakDedhiya** [comment]
yes, Length is not 128 but 512 and that the cause of problem.
Can you recommend which models worked best for you ??
I tried using tt1.pb provided in repo but its not giving identical result as got from python deep-sort implementation.
I was trying to run faceNet model so then i can write similar code for using reidentification model as in python deep-sort implementation.

**Comment by oylz** [comment]
I use deepsort model best(tt1.pb:which I translated from  original checkpoint file in deepsort).In you case,Can you show detail data&result?

**Comment by oylz** [comment]
1 all the codes are translated from original python version in a low level implemention，nearly close，very small difference.
2 of course, you can use any feature getter，remember the feature size is 128.
3 yes,as like [1]，it cached 100 history features every id, calculate by cosine metric and select the best id.
4 no

**Comment by RonakDedhiya** [comment]
Thanks for quick response!!!
I am currently closing the issue. Would raise issue if got stuck.
